Branch xingyu dg contribution

Choo-Xing-Yu opened the pull request:
Add 
1) Adding items and Delivery under implementation in DG
2) AddItemActivityDiagram
3) CommandHistoryTraversalSequenceDiagram
4) Adding an item manual test cases

Choo-Xing-Yu committed:
Merge branch 'master' into branch-xingyu-DG-contribution

Choo-Xing-Yu committed:
Add CommandHistory sequence diagram, AddItemActivityDiagram

Update DG Implementation Add and CommandHistoryTraversal

Choo-Xing-Yu committed:
Add manual test cases for Add Item

Choo-Xing-Yu committed:
Update Command History Traversal

Choo-Xing-Yu committed:
Add newline EOF for PUML, update portfolio

Wincenttjoi commented on docs/diagrams/AddItemActivityDiagram.puml:
```diff
@@ -0,0 +1,10 @@
+@startuml
+(*)  --> "User Executes AddItemCommand"
+If "Item exists in InventoryBook" then
```
If I remember correctly, all the conditions need a [ ... ] right? 
And maybe instead of having true or false like this, it can be replaced as: Item exists in InventoryBook vs Item does not exist in InventoryBook

Wincenttjoi commented on docs/DeveloperGuide.md:
```diff
@@ -153,6 +153,24 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+![AddItemActivityDiagram](images/AddItemActivityDiagram.png)
+
+We've repurposed the `Add` command from AB3 to add on to existing item's quantity instead of throwing an error.
```
```suggestion
Apart from adding a new Item, should there be an existing item in the Inventory Book, `Add` will be able to increase the quantity of that particular Item.
```

Wincenttjoi commented on docs/DeveloperGuide.md:
```diff
@@ -153,6 +153,24 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+The other variables such as:
+
+a) `Metric` `MaxQuantity`
+- Are not allowed to be defined if there exist the same item inside InventoryBook.
```
```suggestion
- Are not allowed to be defined if there exists the same item inside InventoryBook.
```

Wincenttjoi commented on docs/DeveloperGuide.md:
```diff
@@ -153,6 +153,24 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+- Are not allowed to be defined if there exist the same item inside InventoryBook.
+
+b) `Tags`
+- Will be combined together if there exist the same item inside InventoryBook.
```
```suggestion
- Will be combined together if there exists the same item inside InventoryBook.
```

Choo-Xing-Yu replied:
Thanks! Sorry for the oversight

Wincenttjoi commented on docs/DeveloperGuide.md:
```diff
@@ -153,6 +153,24 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+b) `Tags`
+- Will be combined together if there exist the same item inside InventoryBook.
+
+Note: No 2 deliveries are considered the same. Reason being the same person can make multiple delivery orders.
```
No 2 can sound like number 2. And I don't really understand what is the message you're trying to bring across, perhaps a paraphrasing is needed?

Choo-Xing-Yu replied:
Paraphrased!

Wincenttjoi commented on docs/DeveloperGuide.md:
```diff
@@ -164,6 +182,11 @@ method call to return `commandHistory`'s 2nd last command instead of the last co
+
+![CommandHistoryTraversalSequenceDiagram](images/CommandHistoryTraversalSequenceDiagram.png)
+
+The previous command by the user will be reflected in the GUI.
```
Would you want to explain or provide a brief explanation of the diagram?

Choo-Xing-Yu replied:
Added!

Wincenttjoi commented on docs/DeveloperGuide.md:
```diff
@@ -477,6 +500,15 @@ testers are expected to do more *exploratory* testing.
 1. _{ more test cases …​ }_
 
+### Adding an item
+1. Adding an item
```
I think it'll be even better if you split up into 2 sections in this adding part: Adding to an existing item, adding to a non-existing item (instead of explaining them in the expected part)

Choo-Xing-Yu replied:
Done! That's a good idea thank you!

Choo-Xing-Yu committed:
Update docs/DeveloperGuide.md

Co-authored-by: Wincent Tjoi <46394439+Wincenttjoi@users.noreply.github.com>

Choo-Xing-Yu committed:
Update docs/DeveloperGuide.md

Co-authored-by: Wincent Tjoi <46394439+Wincenttjoi@users.noreply.github.com>

Choo-Xing-Yu committed:
Fix AddItemActivityDiagram

Add description to CommandHistoryActivityDiagram

Split add manual test cases to existing/non-existing

Wincenttjoi commented on docs/DeveloperGuide.md:
```diff
@@ -153,6 +153,26 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+Note: Deliveries are all considered unique. Reason being the same person can make multiple delivery orders.
+
+For example, `John` with the address `Choa Chu Kang Block 259` is able to make multiple orders before his previous deliveries are fulfilled.
```
U can use the information source for this kind of note to make it consistent

Choo-Xing-Yu replied:
Added! Thanks!

Wincenttjoi commented on docs/DeveloperGuide.md:
```diff
@@ -477,6 +504,26 @@ testers are expected to do more *exploratory* testing.
+       Expected: Item with `Name` of Chicken, `Quantity` of 123 and `Supplier` of NTUC added 
+       
+    1. Test Case: `add-i n/Chicken q/123 s/giant max/500 metric/kg`
+       Expected: Item with `Name` of Chicken, `Quantity` of 123, `Supplier` of NTUC, `MaxQuantity` of 500 and `Metric` of kg added if there is no existing same item. 
```
```suggestion
       Expected: Item with `Name` of Chicken, `Quantity` of 123, `Supplier` of NTUC, `MaxQuantity` of 500 and `Metric` of kg added when there is no existing same item. 
```

Choo-Xing-Yu replied:
Remedied! Good catch!

Wincenttjoi commented:
![image](https://user-images.githubusercontent.com/46394439/97104736-1e869d00-16f1-11eb-8e74-4af9f3a166c9.png)
This part is abit misleading, update an item sounds like editing

Choo-Xing-Yu committed:
Update DG

Choo-Xing-Yu commented:
> ![image](https://user-images.githubusercontent.com/46394439/97104736-1e869d00-16f1-11eb-8e74-4af9f3a166c9.png)
> This part is abit misleading, update an item sounds like editing

Updated to Adds instead of Updates

Choo-Xing-Yu committed:
Merge branch 'master' into branch-xingyu-DG-contribution

Choo-Xing-Yu committed:
Merge branch 'master' into branch-xingyu-DG-contribution

Wincenttjoi reviewed (approved):
LGTM
